Update DogsbodyTechnology rules

dogsbody opened the pull request.

bardiharborow commented:
@dogsbody Hey! As a site owner, we'd like you to consider [HSTS preloading](https://hstspreload.org/) your domains. Preloading provides a stronger security guarantee than HTTPS-Everywhere, removes the need for a ruleset on our end, and ensures that even users who don't use HTTPS-Everywhere are secure.

koops76 commented:
@dogsbody Recommended reading: https://wiki.mozilla.org/Security/Guidelines/Web_Security.

dogsbody commented:
@bardiharborow Totally agree with HSTS preloading.  We currently have HSTS headers on all our SSL sites however we don't yet serve *all* subdomains over HTTPS due to some legacy issues so can't yet add our domain to HSTS preloading.
One of the reasons we renamed our file in this pull request is that we hope to move our company to a new SSL only domain soon which means we'll finally be able to add it to HSTS preloading :-)

J0WI commented on src/chrome/content/rules/DogsbodyTechnologyLtd.xml:
```diff
@@ -0,0 +1,8 @@
+	<target host="dogsbodyhosting.net" />
+	<target host="www.dogsbodytechnology.com" />
+	<target host="www.dogsbodyhosting.net" />
+        <target host="portal.dogsbodytechnology.com" />
```
Please use only tabs to indent lines.

See also:
````
alerts.dogsbodytechnology.com
pink.dogsbodyhosting.net
purple.dogsbodyhosting.net

jeremyn commented:
@dogsbody Are you still interested in working on this?

dogsbody commented:
@jeremyn apologies, I missed the update from @J0WI 
I have updated the branch to use tabs as suggested.

Regarding:  
> alerts.dogsbodytechnology.com
> pink.dogsbodyhosting.net
> purple.dogsbodyhosting.net
These are servernames which are independent of the https/non-https services that maybe running on them and alerts is in the process of being decommissioned.

Thank you.  Please let me know if there is anything else that needs to happen :-)

jeremyn commented:
@dogsbody The way our rulesets generally work is to try to get a sort of snapshot of the public-facing subdomains under any given domain at the time we create or merge the ruleset. You're welcome to submit updates to a ruleset later on if the situation changes, for example once some servers have been decommissioned etc.

That said, here's what I think the ruleset should look like:

Filename: `Dogsbody.com.xml`
```xml
<!--
	Invalid certificate:
		dogsbody.com
		www.dogsbody.com
		purple.dogsbodyhosting.net

-->
<ruleset name="Dogsbody">
	<target host="status.dogsbody.com" />

	<target host="dogsbodytechnology.com" />
	<target host="www.dogsbodytechnology.com" />
	<target host="alerts.dogsbodytechnology.com" />

	<target host="dogsbodyhosting.net" />
	<target host="www.dogsbodyhosting.net" />
	<target host="pink.dogsbodyhosting.net" />
	<target host="purple-secure.dogsbodyhosting.net" />
	<target host="secure.dogsbodyhosting.net" />

	<rule from="^http:" to="https:" />
</ruleset>
```

Please change your ruleset to match this, and rename the file as well, or let me know if you have any questions.

jeremyn commented:
@dogsbody Thanks for your update, however your new version misses some items from my previous comment. For example you dropped the top comment, didn't change the `ruleset` `name`, and don't sort the subdomains the way we prefer. Can you please make a new update that more closely matches what I suggested?

dogsbody commented:
@jeremyn Thank you for this.  I have updated the ruleset as I believe it should be.  To take your points...

- I have kept the filename and ruleset name as "Dogsbody Technology Ltd." as this is our legal company name and I feel this is an important validation showing users who the legal entity is behind these hostnames.  
- I was basing the ruleset name on what others appear to have done in this repo when their company spans multiple domains.  I will happily spilt these into a separate file per domain if that is required.
- I cannot add our "colour" servers as sites can and do move around on these servers on a daily basis and there is no guarantee that servers running HTTPS one day will be the next.
- I have however done a full audit and added all public URL's that utilise HTTPS including hostnames that are only used for redirects of lost traffic.

I hope this puts things in order.  Thank you again for your help.

jeremyn commented:
@dogsbody Please understand that anyone can submit PRs, and generally individuals or organizations don't "own" rulesets related to them. So even if I merged your PR as-is, someone else could come along later and submit an update, and that can be merged without you knowing.

Also, while we don't want to deliberately antagonize site owners, this project is aimed at helping our users, not site owners. So for example while you might prefer "Dogsbody Technology Ltd." as your legal name, I think that "Dogsbody" is easier to identify. (See https://github.com/EFForg/https-everywhere/pull/12829#issuecomment-338031036 for a related discussion.) If you want, we can split this ruleset into three PRs, one with a filename `DogsbodyTechnology.com.xml` and `ruleset` `name` of `dogsbodytechnology.com` (or `DogsbodyTechnology.com`), another for `dogsbodyhosting.net`, and another for `dogsbody.com`. They would need references to one another, for example see [`ACLU.xml`](https://github.com/EFForg/https-everywhere/blob/203375992d37e878d585e49d050fea283f51e390/src/chrome/content/rules/ACLU.xml) and [`ACLU-of-Arizona.xml`](https://github.com/EFForg/https-everywhere/blob/203375992d37e878d585e49d050fea283f51e390/src/chrome/content/rules/ACLU-of-Arizona.xml). Feel free to pick which one is the main ruleset.

I'm not totally sure I understand the issue with the color domains. Do you mean that one day `pink.dogsbodyhosting.net` could work with HTTPS and the next day it might have some problem?

dogsbody commented:
@jeremyn 

> So for example while you might prefer "Dogsbody Technology Ltd." as your legal name, I think that "Dogsbody" is easier to identify.
If this what you wish then I am more than happy with this entry being called Dogsbody :-)

> I'm not totally sure I understand the issue with the color domains. Do you mean that one day pink.dogsbodyhosting.net could work with HTTPS and the next day it might have some problem?
That is correct :-)

jeremyn commented:
Okay, here's where we should be:

Filename: `Dogsbody.com.xml`
```xml
<!--
    Invalid certificate:
        dogsbody.com
        www.dogsbody.com
 
    Occasional HTTPS errors (see https://github.com/EFForg/https-everywhere/pull/12750):
        pink.dogsbodyhosting.net
        purple.dogsbodyhosting.net
 
--> 
<ruleset name="Dogsbody">
    <target host="status.dogsbody.com" />
 
    <target host="dogsbodytechnology.com" />
    <target host="www.dogsbodytechnology.com" />
    <target host="alerts.dogsbodytechnology.com" />
    <target host="portal.dogsbodytechnology.com" />
    <target host="status.dogsbodytechnology.com" />
    <target host="utils.dogsbodytechnology.com" />
        <test url="http://utils.dogsbodytechnology.com/ADD-TEST-HERE" />
 
    <target host="dogsbodyhosting.net" />
    <target host="www.dogsbodyhosting.net" />
    <target host="pink.dogsbodyhosting.net" />
    <target host="purple-secure.dogsbodyhosting.net" />
    <target host="secure.dogsbodyhosting.net" />
    <target host="status.dogsbodyhosting.net" />
 
    <rule from="^http:" to="https:" />
</ruleset>
```

Your current version has some sorting problems that my version fixes. See [here](https://github.com/EFForg/https-everywhere/blob/ce69f3aa09828e4f94b5b90cd97fa819970039b7/CONTRIBUTING.md#target-ordering) in `CONTRIBUTING.md` for more information.

For `utils.dogsbodytechnology.com`, you added that so I assume it's valid, however the root URL https://utils.dogsbodytechnology.com returns a `403`. In that case we like to have a publicly-available `test` `url` that returns either `2XX` or `3XX`. Can you add an appropriate URL where I've added the `test` tag?

I note that you keep `purple-secure.dogsbodyhosting.net` as a `target` even though it's a "color". Also, I've re-added `secure.dogsbodyhosting.net`.

dogsbody commented:
You know what.  Cancel this pull request.  Thank you for all your help but we'll stick with HSTS and the other technologies for now.  You can cancel this pull request :-)

jeremyn commented:
@dogsbody Please feel free to close the PR if you like.
